dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -6,6 +6,41 @@
 	<Import Project="Xamarin.Shared.Sdk.DefaultItems.targets" />
 	<Import Project="Xamarin.Shared.Sdk.TargetFrameworkInference.targets" />
 
+	<!-- Inject our custom logic into *DependsOn variables -->
+	<PropertyGroup>
+		<BuildDependsOn>
+			$(BuildDependsOn);
+			_CreateAppBundle;
+		</BuildDependsOn>
+
+		<!-- We re-use ComputeFilesToPublish & CopyFilesToPublishDirectory to copy files to the .app -->
+		<CreateAppBundleDependsOn>
+			ComputeFilesToPublish;
+			_ComputePublishLocation;
+			CopyFilesToPublishDirectory;
+		</CreateAppBundleDependsOn>
+	</PropertyGroup>
+
+	<Target Name="_ComputePublishLocation" DependsOnTargets="_GenerateBundleName">
+		<!-- Put .dll, .pdb, .exe and .dylib in the .app -->
+		<PropertyGroup>
+			<_AssemblyPublishDir Condition="'$(_PlatformName)' != 'macOS'">$(MSBuildProjectDirectory)$(_AppBundlePath)\</_AssemblyPublishDir>
+			<_AssemblyPublishDir Condition="'$(_PlatformName)' == 'macOS'">$(MSBuildProjectDirectory)$(_AppBundlePath)\Contents\MonoBundle\</_AssemblyPublishDir>
+			<_DylibPublishDir Condition="'$(_PlatformName)' != 'macOS'">$(MSBuildProjectDirectory)$(_AppBundlePath)\</_DylibPublishDir>
+			<_DylibPublishDir Condition="'$(_PlatformName)' == 'macOS'">$(MSBuildProjectDirectory)$(_AppBundlePath)\Contents\MonoBundle\</_DylibPublishDir>
```

Comment:
Could `_DylibPublishDir` be made equal to `_AssemblyPublishDir` and avoid computing the value twice ?

Also maybe we should re-consider the assembly location on macOS to be identical to the other OS we support ? (out of scope for this PR).

Reply:
> Also maybe we should re-consider the assembly location on macOS to be identical to the other OS we support ? (out of scope for this PR).

The .app structure (and requirements) are different between macOS and the other OSes we support. In particular I believe we can't put files in the root directory on macOS (we should confirm this though).

> Could _DylibPublishDir be made equal to _AssemblyPublishDir and avoid computing the value twice ?

This will eventually all be re-written in the future (there is no fixed assembly directory: they can go in subdirectories (.monotouch-32|64), in frameworks, in the root directory, for app extensions they can be in the container app's root directory, in the container app's frameworks, etc.)

_DylibPublishDir will likely stay the same though.

Reply:
`In particular I believe we can't put files in the root directory on macOS (we should confirm this though).` - 95% sure this is correct.